src/ConstructiveSolidGeometry/Transformations.jl

```diff
@@ -1,69 +1,10 @@
-struct ScaledGeometry{T,P<:AbstractGeometry{T}} <: AbstractTransformedGeometry{T}
-    p::P
-    inv_s::SVector{3,T}
-    ScaledGeometry(p::P, s::SVector{3,T}) where {T,P} = new{T,P}(p, inv.(s))
-end
-in(p::CartesianPoint, g::ScaledGeometry) = in(CartesianPoint(p .* g.inv_s), g.p)
-in(p::CylindricalPoint, g::ScaledGeometry) = in(CartesianPoint(p), g)
-scale(g::AbstractGeometry{T}, s::SVector{3,T}) where {T} = (s == SVector{3,T}(1,1,1) ? g : ScaledGeometry(g, s))
-scale(g::ScaledGeometry{T}, s::SVector{3,T}) where {T} = (inv.(g.inv_s) .* s == SVector{3,T}(1,1,1) ? g.p : ScaledGeometry(g.p, inv.(g.inv_s) .* s))
-#(*)(g::AbstractGeometry{T}, s::SVector{3,T}) where {T} = scale(g, s)
-get_plot_points(sg::ScaledGeometry{T}; n = 30) where {T} = scale!(get_plot_points(sg.p, n = n), inv.(sg.inv_s))
+transform(csg::G, t) where {G <: AbstractConstructiveGeometry} = G(transform(csg.a, t), transform(csg.b, t))
+transform(csg::G, t::Missing) where {G <: AbstractConstructiveGeometry} = csg
 
-function Dictionary(g::ScaledGeometry{T}) where {T}
-    dict = Dictionary(g.p)
-    s = inv.(g.inv_s)
-    if s[1] != 1 dict["x"] = s[1] end
-    if s[2] != 1 dict["y"] = s[2] end
-    if s[3] != 1 dict["z"] = s[3] end
-    OrderedDict{String,Any}("scale" => dict)
-end
+transform(g::AbstractPrimitive, v::AbstractVector) =
+    length(v) > 1 ? transform(transform(g, v[end]), v[1:end-1]) : transform(g, v[1])
```

Comment:
I feel like using `reduce` might be fast here... Something like 
`reduce(v -> transform(g,v), v[2:end], init = v[1])`?

Reply:
Your code throws an error. I think reduce is also meant for binary operations (?). 

I tested my version with 5 transformation and it took 450ns. 
We only need this in the read-in of the configuration file anyhow. Thus, it is fast enough I guess. 
But feel free to play around.

Reply:
will this cover cases where we first translate, then rotate, and then translate again?

Reply:
This should not be handled here but in the read-in when we fill `v`.

This is done by `parse_CSG_transformation` or where this function is get called.
We will have to fix and document in which order we read in transformations which are in the same level in the config file.

Reply:
Sure, we agreed on first rotate and then translate.
But what happens if it gets rotated again at a higher level? Will it be possible to implement this?

Reply:
I believe this already works. But, than it is a rotation around the origin (so not around its own center if it was translated previously). I think this is how it should be. 
But, yes on one level, it should always first be rotation and than translation. There, I am not sure if it implemented correctly at the moment.

Reply:
This works: `transform(g::AbstractPrimitive, v::AbstractVector) = reduce(transform, reverse(v), init = g)`
It has half the allocations as is around 40% faster

Reply:
Sorry, but now the transformations are no longer passed as a vector.
See commit: https://github.com/JuliaPhysics/SolidStateDetectors.jl/pull/163/commits/8005bc0967fcd3021224e709e8fc65db0b8a1e0b